libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2308,19 +2308,19 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 			solAssert(_from.baseType() == _to.baseType(), "");
 			ABIFunctions abi(m_evmVersion, m_revertStrings, m_functionCollector);
 			return Whiskers(R"(
-				function <functionName>(slot) -> memptr {
-					memptr := <allocateTemp>()
-					let end := <encode>(slot, memptr)
-					mstore(<freeMemoryPointer>, end)
+				function <functionName>(slot) -> memPtr {
+					memPtr := <allocateUnbounded>()
+					let end := <encode>(slot, memPtr)
+					<finalizeAllocation>(memPtr, sub(end, memPtr))
```

Comment:
Why did `sub(end, memPtr)` replace `end`? Are we storing the length instead of absolute offset in freeMemPtr now?

Reply:
`mstore(<freeMemoryPointer>, end)` stores the new value of the free memory pointer, i.e. a memory location.`finalizeAllocation` on the other hand takes the current value of the free memory pointer and the size of the allocated memory. Internally it adds the two and passes them to `mstore()`.